File: backend-ms1/src/main/java/ma/zs/stocky/service/facade/admin/commun/ModePaiementAdminService.java
package ma.zs.stocky.service.facade.admin.commun;

import java.util.List;
import ma.zs.stocky.bean.core.commun.ModePaiement;
import ma.zs.stocky.dao.criteria.core.commun.ModePaiementCriteria;
import ma.zs.stocky.zynerator.service.IService;

import ma.zs.stocky.ws.dto.commun.ModePaiementDto;
import org.springframework.http.HttpEntity;


public interface ModePaiementAdminService extends  IService<ModePaiement,ModePaiementCriteria>  {



    HttpEntity<byte[]> createPdf(ModePaiementDto dto) throws Exception;

}

File: frontend/src/app/controller/service/admin/commun/ModePaiementAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import { RoleService } from 'src/app/zynerator/security/Role.service';
import {environment} from 'src/environments/environment';

import {ModePaiementDto} from 'src/app/controller/model/commun/ModePaiement.model';
import {ModePaiementCriteria} from 'src/app/controller/criteria/commun/ModePaiementCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class ModePaiementAdminService extends AbstractService<ModePaiementDto, ModePaiementCriteria> {
     constructor(private http: HttpClient, private roleService: RoleService) {
        super();
        this.setHttp(http);
        this.setApi(environment.apiUrl + 'admin/modePaiement/');
    }

    public constrcutDto(): ModePaiementDto {
        return new ModePaiementDto();
    }

    public constrcutCriteria(): ModePaiementCriteria {
        return new ModePaiementCriteria();
    }
}
